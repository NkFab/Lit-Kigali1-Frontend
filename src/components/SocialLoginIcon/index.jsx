import React from 'react';
import PropTypes from 'prop-types';

const SocialLoginIcon = ({
  id,
  icon,
  alt,
  href,
  dataTest,
}) => (
  <div id={id} data-test={dataTest.div}>
    <a href={href} data-test={dataTest.a}>
      <img src={icon} alt={alt} data-test={dataTest.img} />
    </a>
  </div>
);

SocialLoginIcon.propTypes = {
  id: PropTypes.string.isRequired,
  icon: PropTypes.any.isRequired,
  alt: PropTypes.string.isRequired,
  href: PropTypes.string.isRequired,
  dataTest: PropTypes.object.isRequired,
};

export default SocialLoginIcon;
