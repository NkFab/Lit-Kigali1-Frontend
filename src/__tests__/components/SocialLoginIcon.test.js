import React from 'react';
import { shallow } from 'enzyme';
import SocialLoginIcon from '../../components/SocialLoginIcon';

let component;
const dataTest = {
  div: 'container',
  a: 'image',
  img: 'img',
};
describe('<Login />', () => {
  beforeEach(() => {
    component = shallow(<SocialLoginIcon id="id" alt="alt" href="href" icon="icon" dataTest={dataTest} />);
  });
  test('should render a div', () => {
    const div = component.find(`[data-test='${dataTest.div}']`);
    expect(div.length).toEqual(1);
  });
  test('should render a anchor tag', () => {
    const anchor = component.find(`[data-test='${dataTest.a}']`);
    expect(anchor.length).toEqual(1);
  });
  test('should render an image', () => {
    const img = component.find(`[data-test='${dataTest.img}']`);
    expect(img.length).toEqual(1);
  });
});
