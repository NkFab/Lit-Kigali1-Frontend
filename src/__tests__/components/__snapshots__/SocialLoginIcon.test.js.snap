// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should render a div 1`] = `ShallowWrapper {}`;

exports[`<Login /> should render the <Login /> 1`] = `ShallowWrapper {}`;
